scripts/build.sh

```diff
@@ -2,13 +2,31 @@
 
 set -e
 
-if [[ "$TRAVIS_PULL_REQUEST" == "false" && -z "$TRAVIS_TAG" ]] ; then
-    echo "Git commit is not a pull request or it does not have a release tag so acceptance tests will not run."
-    make build
+function makeRelease {
+    tests/clean.sh
+    make testacc
+    make release
+}
+
+
+if [[ "$TRAVIS_PULL_REQUEST" == "true" ]] ; then
+    echo "Git commit is a pull request - building and running the acceptance tests"
+    makeRelease
+    exit 0
+fi
+
+if [[ -n "$TRAVIS_TAG" ]] ; then
+    echo "Git commit has a release tag - building and running the acceptance tests"
+    makeRelease
+    exit 0
+fi
+
+if [[ "$TRAVIS_BRANCH" == "dev" || "$TRAVIS_BRANCH" == "master" ]] ; then
+    echo "Git commit is on dev or master branch - building and running the acceptance tests"
+    makeRelease
     exit 0
 fi
 
-tests/clean.sh
-make testacc
-make release
+echo "Git commit is not on dev or master branch or it is not a pull request or it does not have a release tag so acceptance tests will not run."
```

Comment:
I understand this case would mainly be for pushed commits mevansam/terraform-provider-cf branches not yet submitted as a PR, for which we want to avoid triggering an acceptance test by default, and rather wait for the branch to the associated with a PR to start testing it.

Suggesting to reword the echoed message to 
"Git commit is likely to be preparation work for an upcoming PR so acceptance tests will not run (not on dev or master branch, nor it is not a pull request, nor it does not have a release tag)"